src/db.py

```
@@ -16,13 +16,36 @@
+        dbapi_connection: Any, _connection_record: Any
+    ) -> None:
+        cursor = dbapi_connection.cursor()
+        try:
+            cursor.execute(
+                "SET hnsw.iterative_scan = %s",
+                (settings.DB.HNSW_ITERATIVE_SCAN,),
+            )
+        finally:
+            cursor.close()
+
+    event.listen(engine.sync_engine, "connect", _set_hnsw_iterative_scan)
```

Ruff (0.16.2)

[error] Undefined name `engine`

(F821)